src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,449 @@
+-- isEmpty ('vertex' x)                      == False
+-- isEmpty ('removeVertex' x $ 'vertex' x)     == True
+-- isEmpty ('removeEdge' 1 2 $ 1 * 2)        == False
+-- @
+empty :: AdjacencyMap a
+empty = coerce AM.empty
+
+-- | Construct the graph comprising /a single isolated vertex/.
+-- Complexity: /O(1)/ time and memory.
+--
+-- @
+-- 'isEmpty'     (vertex x) == False
+-- 'hasVertex' x (vertex x) == True
+-- 'vertexCount' (vertex x) == 1
+-- 'edgeCount'   (vertex x) == 0
+-- @
+vertex :: a -> AdjacencyMap a
+vertex = coerce AM.vertex
+
+-- | Construct the graph comprising a given list of isolated vertices.
+-- Complexity: /O(L * log(L))/ time and /O(L)/ memory, where /L/ is
+-- the length of the given list.
+--
+-- @
+-- vertices []              == 'empty'
+-- vertices [x]             == 'vertex' x
+-- 'hasVertex' x . vertices   == 'elem' x
+-- 'vertexCount' . vertices   == 'length' . 'Data.List.nub'
+-- 'vertexSet'   . vertices   == Set.'Set.fromList'
+-- @
+vertices :: (Ord a) => [a] -> AdjacencyMap a
+vertices = coerce AM.vertices
+
+-- | Perform a disjoint overlay of two different acyclic graphs.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- 'isEmpty' (disjointOverlay x y)             == 'isEmpty' x && 'isEmpty' y
+-- 'hasVertex' (Left z) (disjointOverlay x y)  == 'hasVertex' z x
+-- 'hasVertex' (Right z) (disjointOverlay x y) == 'hasVertex' z y
+-- 'vertexCount' (disjointOverlay x y)         >= 'vertexCount' x
+-- 'vertexCount' (disjointOverlay x y)         == 'vertexCount' x + 'vertexCount' y
+-- 'edgeCount'   (disjointOverlay x y)         >= 'edgeCount' x
+-- 'edgeCount'   (disjointOverlay x y)         == 'edgeCount' x + 'edgeCount' y
+-- 'vertexCount' (disjointOverlay 1 2)         == 2
+-- 'edgeCount'   (disjointOverlay 1 2)         == 0
+-- @
+disjointOverlay ::
+     (Ord a, Ord b)
+  => AdjacencyMap a
+  -> AdjacencyMap b
+  -> AdjacencyMap (Either a b)
+disjointOverlay (AAM a) (AAM b) = AAM (AM.overlay (AM.gmap Left a) (AM.gmap Right b))
+
+-- | Perform a disjoint connect of two different acyclic graphs.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- 'isEmpty'     (disjointConnect x y)           == 'isEmpty'   x   && 'isEmpty'   y
+-- 'hasVertex'   (Left z) (disjointConnect x y)  == 'hasVertex' z x
+-- 'hasVertex'   (Right z) (disjointConnect x y) == 'hasVertex' z y
+-- 'vertexCount' (disjointConnect x y)           >= 'vertexCount' x
+-- 'vertexCount' (disjointConnect x y)           == 'vertexCount' x + 'vertexCount' y
+-- 'edgeCount'   (disjointConnect x y)           >= 'edgeCount' x
+-- 'edgeCount'   (disjointConnect x y)           >= 'edgeCount' y
+-- 'edgeCount'   (disjointConnect x y)           >= 'vertexCount' x * 'vertexCount' y
+-- 'edgeCount'   (disjointConnect x y)           == 'vertexCount' x * 'vertexCount' y + 'edgeCount' x + 'edgeCount' y
+-- 'vertexCount' (disjointConnect 1 2)           == 2
+-- 'edgeCount'   (disjointConnect 1 2)           == 1
```

Comment:
As in `disjointOverlay`, some properties here seem to be redundant. This is not too critical: feel free to leave everything as is -- I'll do a minor revision after merging this PR anyway.